wrangler-service/src/main/java/io/cdap/wrangler/service/adls/ADLSHandler.java

```diff
@@ -0,0 +1,382 @@
+            }
+        });
+    }
+
+    /**
+     * A method to fetch a file from an ADLS client
+     *
+     * @param connection
+     * @param fileQueryDetails
+     * @param namespaceID
+     * @throws IOException
+     */
+    private ADLSConnectionSample fetchFileFromClient(Connection connection, FileQueryDetails fileQueryDetails,
+                                                     NamespacedId namespaceID) throws IOException {
+        ADLStoreClient client = initializeAndGetADLSClient(connection);
+        DirectoryEntry file = ADLSUtility.getFileFromClient(client, fileQueryDetails.getFilePath());
+        try (InputStream inputStream = ADLSUtility.clientInputStream(client, fileQueryDetails)) {
+            if (fileQueryDetails.getHeader() != null && fileQueryDetails.getHeader().equalsIgnoreCase
+                    ("text/plain")) {
+                return loadSamplableFile(namespaceID, fileQueryDetails.getScope(), inputStream, file,
+                        fileQueryDetails.getLines(), fileQueryDetails.getFraction(), fileQueryDetails.getSampler());
+            } else return loadFile(namespaceID, fileQueryDetails.getScope(), inputStream, file);
+        } catch (ADLException e) {
+            throw new StatusCodeException(e.getMessage(), e, e.httpResponseCode);
+        }
+    }
+
+    private ADLSConnectionSample loadSamplableFile(NamespacedId connectionId,
+                                                   String scope, InputStream inputStream, DirectoryEntry fileEntry,
+                                                   int lines, double fraction, String sampler) throws IOException {
+        SamplingMethod samplingMethod = SamplingMethod.fromString(sampler);
+
+        if (sampler == null || sampler.isEmpty() || SamplingMethod.fromString(sampler) == null) {
+            samplingMethod = SamplingMethod.FIRST;
+        }
+        final SamplingMethod samplingMethod1 = samplingMethod;
+        try (BoundedLineInputStream blis = BoundedLineInputStream.iterator(inputStream, Charsets.UTF_8, lines)) {
+            String name = fileEntry.name;
+            String file = String.format("%s:%s", scope, fileEntry.name);
+            String fileName = fileEntry.fullName;
+            String identifier = ServiceUtils.generateMD5(file);
+            // Set all properties and write to workspace.
+            Map<String, String> properties = new HashMap<>();
+            properties.put(PropertyIds.ID, identifier);
+            properties.put(PropertyIds.FILE_PATH, fileEntry.fullName);
+            properties.put(PropertyIds.NAME, name);
+            properties.put(PropertyIds.CONNECTION_TYPE, ConnectionType.ADLS.getType());
+            properties.put(PropertyIds.SAMPLER_TYPE, samplingMethod.getMethod());
+            properties.put(PropertyIds.CONNECTION_ID, connectionId.getId());
+            // ADLS specific properties.
+            properties.put("file-name", fileEntry.fullName);
```

Comment:
this has the same value as FILE_PATH. Is it needed?

Reply:
Currently, no. It was needed earlier, I will remove it.